plugins/inputs/tail/tail.go

```diff
@@ -205,16 +240,22 @@ func (t *Tail) tailNewFiles(fromBeginning bool) error {
 
 			// create a goroutine for each "tailer"
 			t.wg.Add(1)
+
 			go func() {
 				defer t.wg.Done()
-				t.receiver(parser, tailer)
+				if t.multiline.IsEnabled() {
+					go t.receiverMultiline(parser, tailer)
```

Comment:
I don't really like that receiverMultiline forks the behavior here. It definitely adds a lot of code complexity and likely introduces bugs in how the two receivers process input. eg, search for "Block until plugin is stopping or room is available to add metrics" and you'll find that this other receiver code has already changed, and multiline isn't supporting coordination with downstream outputs to not overwhelm them. Now the multiline receiver is out of sync and won't respect this behavior. In the best case scenario, future updates will have to update two versions of the code.

How difficult do you think it'd be to merge the two receiver functions? Any concerns?